### ISSUE TITLE
Depreciate caret redirection for fish shell

### ISSUE THREAD

**Issue by caizixian** [cea-hpc/modules#337]
### Describe the bug
Currently, module doesn't work with fish 3.1.0, due to a bug in fish. https://github.com/fish-shell/fish-shell/issues/6591
The bug has been resolved in fish 3.1.1.
However, this behavior will not be compatible with future versions of fish anymore, and therefore should be rewritten
https://github.com/fish-shell/fish-shell/issues/6591#issuecomment-585775178

### To Reproduce
Run `/usr/bin/tclsh /usr/share/Modules/libexec/modulecmd.tcl fish autoinit`, which contains the following output
```fish
function module
   _module_raw $argv ^&1
end
```

### Expected behavior
Rewrite `^&1`, per the suggestion from the upstream https://github.com/fish-shell/fish-shell/issues/6591#issuecomment-585775178

### Modules version and configuration
```console
$ module --version
Modules Release 4.4.1 (2020-01-03)
```

**Comment by xdelaruelle** [comment]
Thanks for your detailed report.

This is a duplicate of issue #325 and the good news is that it is fixed by 711c901, which is included in the newest version of Modules: v4.5.0.

**Comment by caizixian** [comment]
@xdelaruelle sorry for not checking the issue list carefully. I thought it's new and only checked the opened issues.
